[BUG] Invalid schema in CodeCompanion

I sporadically get this bug when adding vectorcode to the context, not exactly sure if this is a CodeCompanion issue or a vectorcode issue. I'll add more details if I can nail down a way to reproduce 

```
[ERROR] 2025-05-13 16:12:50
Error: {
  "error": {
    "message": "Invalid schema for function 'vectorcode': In context=('properties', 'options'), 'required' is required to be supplied and to be an array including every key in properties. Missing 'count'.",
    "type": "invalid_request_error",
    "param": "tools[2].function.parameters",
    "code": "invalid_function_parameters"
  }
}
```

Sometimes I also get this error 

```
[ERROR] 2025-05-14 15:52:27
Error: {
  "error": {
    "message": "Invalid schema for function 'vectorcode': In context=('properties', 'options'), 'required' is required to be supplied and to be an array including every key in properties. Missing 'project_root'.",
    "type": "invalid_request_error",
    "param": "tools[0].function.parameters",
    "code": "invalid_function_parameters"
  }
}
```

With project root. 
I should have the version that has the [project_root related fix](https://github.com/Davidyz/VectorCode/commit/0be48d75aff47c9d29d448766eacb946844bc718). 

```
cat nvim/lazy-lock.json | grep -i vectorcode
  "VectorCode": { "branch": "main", "commit": "0be48d75aff47c9d29d448766eacb946844bc718" },
```

This doesn't seem to fix it for me exactly; looks like maybe a new issue? `minimum is not permitted` 
```
-- Adapter: "OpenAI"
-- Buffer: 6
-- Context: "" (1)

local settings = {
  model = "gpt-4o",  -- "gpt-4.1", "gpt-4o-mini", "gpt-4-turbo-preview", "gpt-4", "gpt-3.5-turbo", "o1-2024-12-17", "o3-2025-04-16", "o3-mini-2025-01-31", "o4-mini-2025-04-16", 
  temperature = 1,
  top_p = 1,
  stop = nil,
  max_tokens = nil,
  presence_penalty = 0,
  frequency_penalty = 0,
  logit_bias = nil,
  user = nil,
}

local messages = 
{ {
    content = "You are an AI programming assistant named \"CodeCompanion\". You are currently plugged into the Neovim text editor on a user's machine.\n\nYour core tasks include:\n- Answering general programming questions.\n- Explaining how the code in a Neovim buffer works.\n- Reviewing the selected code from a Neovim buffer.\n- Generating unit tests for the selected code.\n- Proposing fixes for problems in the selected code.\n- Scaffolding code for a new workspace.\n- Finding relevant code to the user's query.\n- Proposing fixes for test failures.\n- Answering questions about Neovim.\n- Running tools.\n\nYou must:\n- Follow the user's requirements carefully and to the letter.\n- Keep your answers short and impersonal, especially if the user's context is outside your core tasks.\n- Minimize additional prose unless clarification is needed.\n- Use Markdown formatting in your answers.\n- Include the programming language name at the start of each Markdown code block.\n- Avoid including line numbers in code blocks.\n- Avoid wrapping the whole response in triple backticks.\n- Only return code that's directly relevant to the task at hand. You may omit code that isn’t necessary for the solution.\n- Avoid using H1, H2 or H3 headers in your responses as these are reserved for the user.\n- Use actual line breaks in your responses; only use \"\\n\" when you want a literal backslash followed by 'n'.\n- All non-code text responses must be written in the English language indicated.\n- Multiple, different tools can be called as part of the same response.\n\nWhen given a task:\n1. Think step-by-step and, unless the user requests otherwise or the task is very simple, describe your plan in detailed pseudocode.\n2. Output the final code in a single code block, ensuring that only relevant code is included.\n3. End your response with a short suggestion for the next user turn that directly supports continuing the conversation.\n4. Provide exactly one complete reply per conversation turn.\n5. If necessary, execute multiple tools in a single turn.",
    cycle = 1,
    id = 1191187114,
    opts = {
      tag = "from_config",
      visible = false
    },
    role = "system"
  }, {
    content = "vectorcode\n\nwhat is my project doing in the config",
    cycle = 1,
    id = 1821372236,
    opts = {
      visible = true
    },
    role = "user"
  }, {
    content = "### VectorCode, a repository indexing and query tool.\n\n1. **Purpose**: This gives you the ability to access the repository to find information that you may need to assist the user.\n\n2. **Key Points**:\n  - The path of a retrieved file will be wrapped in `<path>` and `</path>` tags. Its content will be right after the `</path>` tag, wrapped by `<content>` and `</content>` tags. Do not include the `<path>``</path>` tags in your answers when you mention the paths.\n  - If you used the tool, tell users that they may need to wait for the results and there will be a virtual text indicator showing the tool is still running\n  - Include one single command call for VectorCode each time. You may include multiple keywords in the command\n  - VectorCode is the name of this tool. Do not include it in the query unless the user explicitly asks\n  - Use the `ls` command to retrieve a list of indexed project and pick one that may be relevant, unless the user explicitly mentioned 'this project' (or in other equivalent expressions)\n  - **The project root option MUST be a valid path on the filesystem. It can only be one of the results from the `ls` command or from user input**\n\n  - If the user did not specify how many documents to retrieve, **start with 10 documents**\n  - If you decide to call VectorCode tool, do not start answering the question until you have the results. Provide answers based on the results and let the user decide whether to run the tool again\n  - **Use at your discretion** when you feel you don't have enough information about the repository or project\n  - **Don't escape** special characters\n  - separate phrases into distinct keywords when appropriate\n  - If a class, type or function has been imported from another file, this tool may be able to find its source. Add the name of the imported symbol to the query\n  - Avoid retrieving one single file because the retrieval mechanism may not be very accurate\n  - When providing answers based on VectorCode results, try to give references such as paths to files and line ranges, unless you're told otherwise (but do not include the full source code context)\n  - VectorCode is the name of this tool. Do not include it in the query unless the user explicitly asks\n  - If the retrieval results do not contain the needed context, increase the file count so that the result will more likely contain the desired files\n  - If the returned paths are relative, they are relative to the root of the project directory\n  - Do not suggest edits to retrieved files that are outside of the current working directory, unless the user instructed otherwise\n  - When specifying the `project_root` parameter when making a query, make sure you run the `ls` tool first to retrieve a list of valid, indexed projects\n  - If a query failed to retrieve desired results, a new attempt should use different keywords that are orthogonal to the previous ones but with similar meanings\n  - Do not use exact query keywords that you have used in a previous tool call in the conversation, unless the user instructed otherwise, or with different count/project_root\n  - The current working directory is /Users/zach/Code/litellm. Assume the user query is about this project, unless the user asked otherwise or queries from the current project fails to return useful results. \n",
    cycle = 1,
    id = 332559912,
    opts = {
      reference = "<tool>vectorcode</tool>",
      tag = "tool",
      visible = false
    },
    role = "system"
  }, {
    content = "",
    cycle = 1,
    id = 257508401,
    opts = {
      visible = true
    },
    role = "user"
  }, {
    content = "",
    cycle = 1,
    id = 257508401,
    opts = {
      visible = true
    },
    role = "user"
  } }
```

```
[ERROR] 2025-05-15 13:20:26
Error: {
  "error": {
    "message": "Invalid schema for function 'vectorcode': In context=('properties', 'options', 'properties', 'count'), 'minimum' is not permitted.",
    "type": "invalid_request_error",
    "param": "tools[0].function.parameters",
    "code": "invalid_function_parameters"
  }
}
```

At https://github.com/Davidyz/VectorCode/commit/fc46284420fce6ac9a2dfd1f9db52dfad2cf8fea, I get this error 

```
[ERROR] 2025-05-17 00:18:50
Error: {
  "error": {
    "message": "Invalid schema for function 'vectorcode': In context=('properties', 'options', 'properties', 'project_root'), 'default' is not permitted.",
    "type": "invalid_request_error",
    "param": "tools[0].function.parameters",
    "code": "invalid_function_parameters"
  }
}
```

--- a/lua/vectorcode/integrations/codecompanion/func_calling_tool.lua
+++ b/lua/vectorcode/integrations/codecompanion/func_calling_tool.lua
@@ -160,10 +160,12 @@ return check_cli_wrap(function(opts)
                 count = {
                   type = "integer",
                   minimum = 1,
+                  default = opts.default_num,
                   description = "Number of documents to retrieve, must be positive",
                 },
                 project_root = {
                   type = "string",
+                  default = "",
                   description = "Project path to search within (must be from 'ls' results). Use empty string for the current project.",
                 },
               },

--- a/lua/vectorcode/integrations/codecompanion/func_calling_tool.lua
+++ b/lua/vectorcode/integrations/codecompanion/func_calling_tool.lua
@@ -159,12 +159,10 @@ return check_cli_wrap(function(opts)
                 },
                 count = {
                   type = "integer",
-                  default = opts.default_num,
                   description = "Number of documents to retrieve, must be positive",
                 },
                 project_root = {
                   type = "string",
-                  default = "",
                   description = "Project path to search within (must be from 'ls' results). Use empty string for the current project.",
                 },
               },

--- a/lua/vectorcode/integrations/codecompanion/func_calling_tool.lua
+++ b/lua/vectorcode/integrations/codecompanion/func_calling_tool.lua
@@ -70,6 +70,9 @@ return check_cli_wrap(function(opts)
             action.options.query = { action.options.query }
           end
           vim.list_extend(args, action.options.query)
+          if action.options.project_root == "" then
+            action.options.project_root = nil
+          end
           if action.options.project_root ~= nil then
             action.options.project_root = vim.fs.normalize(action.options.project_root)
             if
@@ -156,18 +159,19 @@ return check_cli_wrap(function(opts)
                 },
                 count = {
                   type = "integer",
+                  minimum = 1,
                   description = "Number of documents to retrieve, must be positive",
                 },
                 project_root = {
                   type = "string",
-                  description = "Project path to search within (must be from 'ls' results)",
+                  description = "Project path to search within (must be from 'ls' results). Use empty string for the current project.",
                 },
               },
-              required = { "query" },
+              required = { "query", "count", "project_root" },
               additionalProperties = false,
             },
           },
-          required = { "command" },
+          required = { "command", "options" },
           additionalProperties = false,
         },
         strict = true,

--- a/lua/vectorcode/integrations/codecompanion/func_calling_tool.lua
+++ b/lua/vectorcode/integrations/codecompanion/func_calling_tool.lua
@@ -159,7 +159,6 @@ return check_cli_wrap(function(opts)
                 },
                 count = {
                   type = "integer",
-                  minimum = 1,
                   default = opts.default_num,
                   description = "Number of documents to retrieve, must be positive",
                 },

